media: Replace rtcStats singleton with factory function

havardholvik opened the pull request:
### Description

**Summary:**
Replace singleton rtcStats connection object with factory function for better lifecycle control and testability.

**Related Issue:**

<!-- Link to the GitHub issue that this PR addresses, if applicable. -->

### Testing

Unit tests added.

### Checklist

-   [x] My code follows the project's coding standards.
-   [x] Prefixed the PR title and commit messages with the service or package name
-   [x] I have written unit tests (if applicable).
-   [ ] I have updated the documentation (if applicable).
-   [x] By submitting this pull request, I confirm that my contribution is made
        under the terms of the MIT license.

havardholvik committed:
media: Replace rtcStats singleton with factory function

havardholvik commented:
/canary

havardholvik commented:
/canary

kevinwhereby reviewed (approved):
Added all the piping to init this in `bundles/stats.js` and pass it around where required and it all works as expected 🎉

havardholvik commented:
/canary

havardholvik commented:
/canary

havardholvik commented:
/canary

havardholvik committed:
core: Use new major media version, adopt rtcStats integration

Make the rtcConnection slice initialize a fresh rtcConnection before
passing it into the media functions which requires this.
